lib/block-supports/padding.php

```diff
@@ -0,0 +1,59 @@
+<?php
+/**
+ * Padding block support flag.
+ *
+ * @package gutenberg
+ */
+
+/**
+ * Registers the style block attribute for block types that support it.
+ *
+ * @param WP_Block_Type $block_type Block Type.
+ */
+function gutenberg_register_padding_support( $block_type ) {
+	$has_padding_support = gutenberg_block_has_support( $block_type, array( 'spacing', 'padding' ), false );
+
+	// Setup attributes and styles within that if needed.
+	if ( ! $block_type->attributes ) {
+		$block_type->attributes = array();
```

Comment:
I think it's okay to move this check/initialization in the below check.

Reply:
I don't like nested "ifs" too much